### PR TITLE
Adding support for CLI commands over an active MSP port.

### PR DESCRIPTION
**Pull request by blckmn**
App support for the use of CLI commands over an active MSP port.

Firmware required: https://github.com/betaflight/betaflight/pull/13940

### CODE REVIEW

File: src/js/msp.js [hunk 1]
```diff
@@ -305,6 +341,29 @@ const MSP = {
         bufView[bufferSize - 1] = this.crc8_dvb_s2_data(bufView, 3, bufferSize - 1);
         return bufferOut;
     },
+    encode_message_cli(str) {
+        const data = Array.from(str, c => c.charCodeAt(0));
+        const dataLength = data ? data.length : 0;
+        const bufferSize = dataLength + 4;
```
Comment (haslinghuis) [inline src/js/msp.js:347]: Perhaps add a comment why we need magic number 4.

File: src/js/fc.js [hunk 2]
```diff
@@ -798,10 +798,10 @@ const FC = {
     getSupportedSerialRxTypes: () => {
         if (FC.CONFIG.buildOptions?.length) {
             const options = FC.CONFIG.buildOptions;
-            let supportedRxTypes = ['NONE',];
+            let supportedRxTypes = ['NONE'];
```
Comment (haslinghuis) [inline src/js/fc.js:801]: Please rebase. This was already fixed in https://github.com/betaflight/betaflight-configurator/commit/a23f3cfb33018000815291b9b9b505bb32b6d8b9

File: src/js/msp.js [hunk 3]
```diff
@@ -72,9 +82,37 @@ const MSP = {
 
         for (const chunk of data) {
             switch (this.state) {
+            case this.decoder_states.CLI_COMMAND:
+                switch (chunk) {
+                    case this.symbols.END_OF_TEXT:
+                        this.cli_output.push(this.cli_buffer.join(''));
+                        this.cli_buffer.length = 0;
```
Comment (haslinghuis) [inline src/js/msp.js:89]: Should we empty the array instead ?

Comment (blckmn) [reply src/js/msp.js:89]: Any reference to the array remains if you "set" the array using ` = [];`

Setting length to 0 means any reference to the array is also cleared. In this case the buffer should be cleared, and no reference should remain.

https://stackoverflow.com/questions/1232040/how-do-i-empty-an-array-in-javascript

**Review by haslinghuis — APPROVED** [review]
Tested with small implementation:

image

**Review by nerdCopter — APPROVED** [review]
- approving for workflow, untested by me

**Pull request merged**